src/lib/state.ts

```diff
@@ -203,7 +203,7 @@ function createState<T>(): InternalStateType<T> {
       if (!GlobalContext.inCompile() && !GlobalContext.inAnalyze()) {
         let account: Account;
         try {
-          account = Mina.getAccount(address);
+          account = Mina.getAccount(address, getDefaultTokenId());
```

Comment:
could we use `party.body.tokenId` for the token id here as well?

Reply:
Unless I'm mistaken, there is no `party.body.tokenId` in scope here, the only access to a Party here is through `this._contract.instance.self` which doesn't seem right to be referencing the tokenId property.

Reply:
`this._contract.instance.self.body.tokenId` is fine! not pretty but it will reliably hold the party connected to the smart contract for which this is the state. we need to get the state from the account with the right token id